templates/named.conf.erb

```diff
@@ -3,7 +3,9 @@
 include "<%= scope.lookupvar('::dns::rndckeypath') %>";
 
 controls  {
-	inet 127.0.0.1 port 953 allow { 127.0.0.1; } keys { "rndc-key"; };
+<%- scope.lookupvar('::dns::controls_hash').sort_by {|k, v| k}.each do |control_ip, control_hash| -%>
+	inet <%= control_ip %> port <%= control_hash['port'] %> allow { <% control_hash['allowed_addresses'].sort.each do |address| %> <%= address %>; <% end %>} keys { <% control_hash['keys'].sort.each do |keys| %> <%= key %>; <% end %>};
```

Comment:
You're omitting the semi-colons and also the quotes for the keys. I think we need a verify_exact_contents test for named.conf in dns_init_spec.rb.